industrial_taxonomy/pipeline/glass_house/flow.py

```diff
@@ -0,0 +1,196 @@
+            Cos_config,
+            Fuzzy_config,
+            match_names_stream,
+        )
+        from pandas import DataFrame
+        from toolz.curried import curry, pipe
+        from toolz.curried import map as tmap
+
+        from utils import optimise_chunk
+
+        starmap = curry(starmap_)
+
+        def persist_chunk(chunk_id: int, chunk: DataFrame) -> int:
+            key = f"match-chunks-{chunk_id}"
+            with S3(run=self) as s3:
+                s3.put(key, chunk.to_csv())
+            print(f"Persisted chunk {chunk_id} of shape {chunk.shape}")
+            return chunk_id
+
+        tmp_dir = Path(TemporaryDirectory(dir=".").name)
+        tmp_dir.mkdir()
+
+        cos_config = Cos_config()
+        fuzzy_config = Fuzzy_config(num_perm=128)
+        match_config = dict(
+            threshold=50,
+            chunksize=100_000,
+            cos_config=cos_config,
+            fuzzy_config=fuzzy_config,
+            tmp_dir=tmp_dir,
+        )
+
+        self._chunk_ids: List[int] = pipe(
+            match_names_stream(
+                [self.clean_names_x.tolist(), self.clean_names_y.tolist()],
+                **match_config,
+            ),
+            tmap(optimise_chunk),
+            enumerate,  # Give chunks ID's
+            starmap(persist_chunk),
+            list,  # Realise computation
+        )
+        self.next(self.extract_top_matches)
+
+    @resources(memory=64_000)
+    @step
+    def extract_top_matches(self):
+        """Extract top matches."""
+        from io import StringIO
+        from pandas import concat, DataFrame, read_csv
+        from utils import get_top_matches
+
+        def get_chunk(chunk_id: int) -> DataFrame:
+            key = f"match-chunks-{chunk_id}"
+            try:
+                with S3(run=self) as s3:
+                    return read_csv(StringIO(s3.get(key).text))
+            except MetaflowS3NotFound:  # In case of Flow resumption
+                with S3(run=Run(f"{current.flow_name}/{current.origin_run_id}")) as s3:
+                    return read_csv(StringIO(s3.get(key).text))
+
+        self.top_matches = get_top_matches(
+            concat(map(get_chunk, self._chunk_ids)).reset_index(drop=True)
+        )
+
+        print("Number of matches: ", self.top_matches.shape[0])
+        self.next(self.end)
+
+    @step
+    def end(self):
+        """Merge names and id's back in and convert to dict."""
+
+        self.full_top_matches = (
+            self.top_matches.merge(
+                self.names_y.reset_index().rename(columns={"index": "index_y"}),
+                left_on="y",
+                right_index=True,
+                validate="1:1",
+            )
+            .merge(
+                self.names_x.reset_index().rename(columns={"index": "index_x"}),
+                left_on="x",
+                right_index=True,
+                validate="m:1",
+            )
+            .rename(columns={"name_x": "company_name", "name_y": "org_name"})
```

Comment:
Partly addresses comment about company names above above although I would change `org_name` to `glass_name` to avoid potential confusions about the source.

Reply:
`org_name` was just staying consistent with the existing Glass data terminology.

If we change `org_name -> glass_name` here then we'll need to do the same across all the existing Glass flows/getters...

Reply:
Let's not go there...